packages/api/cms-api/src/auth/strategies/basic-auth.strategy.ts

```diff
@@ -1,17 +1,23 @@
-import { forwardRef, Inject, Injectable } from "@nestjs/common";
+import { Injectable } from "@nestjs/common";
 import { PassportStrategy } from "@nestjs/passport";
 import { BasicStrategy } from "passport-http";
 
-import { AUTH_MODULE_CONFIG } from "../auth.constants";
-import { AuthModuleConfig } from "../auth.module";
+export interface AuthBasicAuthStrategyConfig {
+    apiPassword: string;
+}
 
-@Injectable()
-export class BasicAuthStrategy extends PassportStrategy(BasicStrategy, "basic") {
-    constructor(@Inject(forwardRef(() => AUTH_MODULE_CONFIG)) private readonly config: AuthModuleConfig) {
-        super();
-    }
+// eslint-disable-next-line @typescript-eslint/explicit-function-return-type
+export function createBasicAuthStrategy(config: AuthBasicAuthStrategyConfig) {
```

Comment:
coud be named (now that we get incompatible anyway) ApiPasswordAuthStrategy

Reply:
I don't like the term ApiPassword, it's too unclear what's actually meant. I suggest to use BasicAuthStrategy und rename the environment variable.

Reply:
Using a single Term is good! And removing that from demo is also good, now with AuthModule in app so easy.

But BasicAuth doesn't imply a fixed static user/password - it's just about the how the auth data is transmitted.

What about:
- StaticBasicAuth
- BasicAuthStatic
- BasicAuthApiPassword

(oh, and imho you could remove the default "vivid")

Reply:
I (hopefully) improved the naming, now Comet provides these strategies:
- StaticCredentialsBasicStrategy
- AuthProxyJwtStrategy
- StaticAuthedUserStrategy